semantics/src/main/scala/aqua/semantics/CompilerState.scala

```diff
@@ -26,6 +27,16 @@ object CompilerState {
       types = TypesState.init[F](ctx)
     )
 
+  implicit def semigroupState[S[_]]: Semigroup[CompilerState[S]] =
```

Comment:
Please add TODO to remove it after switching from `RawContext` to `LSPContext[S]`. This `Semigroup` is not real, it's super dangerous, as it mixes non-isolated intermediary states during semantic analysis.